### PR TITLE
Frontend setup

### PR DESCRIPTION
**Pull request by CarlosM10011**
This pull request fixes a few bugs that were present with the build scripts, implements initial JSAction support, and adds placeholder code for testing the functionality of the build process.

### CODE REVIEW

**Comment by Lorraine-Lyu** [comment]
@googlebot I consent.

**Review by Lorraine-Lyu — APPROVED** [review]

**Review by MikolJiron — APPROVED** [review]
LGTM

**Comment by MikolJiron** [comment]
@googlebot I consent.

**Review by tinagrasso — APPROVED** [review]

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,9 @@
-[submodule "closure-templates"]
+[submodule "deps/closure-templates"]
 	path = deps/closure-templates
 	url = https://github.com/google/closure-templates
 [submodule "deps/closure-library"]
 	path = deps/closure-library
-	url = https://github.com/google/closure-library
+	url = https://github.com/CarlosM10011/closure-library
```
Comment (tinagrasso) [inline .gitmodules:6]: Why did you change where this submodule points to?

**Review by tinagrasso — CHANGES_REQUESTED** [review]

**Review by hzimz — APPROVED** [review]

File: .gitmodules (hunk 1 above)
Comment (CarlosM10011) [reply .gitmodules:6]: I forked it because I changed closure/goog/events/browserfeature.js to use `goog.module`, instead of `goog.provide` and `goog.scope`. The old file using `goog.scope` would cause the compiler to throw a runtime exception with an internal compiler error. I'm not entirely sure why it is gone, but converting the file caused the error to vanish.

I submitted a PR at (https://github.com/google/closure-library/pull/1080).

**Review by tinagrasso — APPROVED** [review]
